miden-crypto/src/merkle/smt/partial.rs

```diff
@@ -39,14 +43,14 @@ impl PartialSmt {
     /// Returns an error if:
     /// - the new root after the insertion of a (leaf, path) tuple does not match the existing root
     ///   (except if the tree was previously empty).
-    pub fn from_proofs<I>(paths: I) -> Result<Self, MerkleError>
+    pub fn from_proofs<I>(proofs: I) -> Result<Self, MerkleError>
     where
         I: IntoIterator<Item = SmtProof>,
     {
         let mut partial_smt = Self::new();
 
-        for (path, leaf) in paths.into_iter().map(SmtProof::into_parts) {
-            partial_smt.add_path(leaf, path)?;
+        for (proof, leaf) in proofs.into_iter().map(SmtProof::into_parts) {
+            partial_smt.add_path(leaf, proof)?;
         }
```

Comment:
nit: is there a reason not to user `PartialSmt::add_proof()` here?

Reply:
No - good point, we can update that!

Reply:
Let's also make sure to update the doc comment.